### PR TITLE
Add pypromice

### PR DESCRIPTION
**Pull request by PennyHow**
This is a PR for the pypromice package to be added. Myself, @patrickjwright and @mankoff have agreed to be maintainers, as discussed [here](https://github.com/GEUS-Glaciology-and-Climate/pypromice/issues/148). However, if they need to also confirm this here then just let us know. 

Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by PennyHow** [comment]
@mankoff, was there a specific reason why an LGPL-3.0 license was adopted for pypromice? Would any in the list above be better alternatives?

**Comment by mankoff** [comment]
No specific reason. I do not have any strong preference among the alternatives.

**Comment by PennyHow** [comment]
Okay. I think [the GPL license](https://choosealicense.com/licenses/gpl-3.0/) may be a better fit, and it looks like it would fit with the licensing conventions used by conda. @patrickjwright, do you have any opinion or experience picking licenses for a package similar to pypromice?

**Comment by patrickjwright** [comment]
I do not! Really, no experience in the licensing world. If you need a second opinion, happy to dive in a bit though.

**Comment by PennyHow** [comment]
My choice would be either the GPL v2 or v3 license. 

The main difference is that v3 grants patent rights to contributors, as shown in the table here: https://choosealicense.com/appendix/ 

My guess would be v2 (without patent rights) is a better choice for us as the patent should be owned by the organisation and not individuals.

In order to integrate a license change that is reflected here, we need to change the license on the Github repo and then make a new release (which will then also make a new release to pypi). The conda build pulls from the pypi build. I'm happy to make the license change and then wait for a patch or change to pypromice that warrants a minor or patch release. 

As far as I can tell, the conda-forge team will not do anything until I specifically call upon them in this issue. So we can take all the time we like with this.

**Comment by PennyHow** [comment]
@ladsmund, please can you confirm here that you would like to be a maintainer of this package?

**Comment by PennyHow** [comment]
As I made a new release of pypromice, I could change the licensing and all checks seems to pass now. @ladsmund @mankoff @patrickjwright, are we all happy? Once I have confirmation from each of you then I'll ping the conda-forge team.

**Comment by mankoff** [comment]
👍

**Comment by patrickjwright** [comment]
👍

**Review by ladsmund — APPROVED** [review]

**Comment by ladsmund** [comment]
👍

**Review by ladsmund — APPROVED** [review]

**Comment by PennyHow** [comment]
@conda-forge/help-python ready for review!

File: recipes/pypromice/meta.yaml
Comment (ocefpaf) [inline recipes/pypromice/meta.yaml:13]: Looks like this package has scripts and cannot be noarch. Can you open an issue upstream and ask them to move to entry_points? Scripts are deprecated and cause all sorts of trouble for downstream packagers.

Comment (ocefpaf) [reply recipes/pypromice/meta.yaml:13]: You can't do that here now. Upstream must convert those scripts to entry_points. See https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#entry-points

For now, we can merge this without noarch but it is highly recommend to change that I'm the future.

File: recipes/pypromice/meta.yaml
Comment (ocefpaf) [inline recipes/pypromice/meta.yaml:38]: Should probably add a simple CLI test for those scripts here.

File: recipes/pypromice/meta.yaml
Comment (PennyHow) [reply recipes/pypromice/meta.yaml:13]: But if it cannot be `noarch` then we will not have cross-platform compatibility, right? 

If this is the case, then we would like to make the upstream changes and move the scripts to `entry_points` before merging.

As pypromice maintainers, we can start moving the scripts as soon as possible.

Comment (ocefpaf) [reply recipes/pypromice/meta.yaml:13]: That's great. Usually folks are eager to get packages in without waiting for upstream changes and new releases.

**Comment by PennyHow** [comment]
@ocefpaf CLI scripts have been converted to `entry_points` in the newest version of pypromice now. We also include these in our pypromice unit testing, but I saw that the new `grayskull` recipe we generated also added some simple CLI testing too. 

Does this resolve the platform compatibility issues now?

**Pull request merged**
